### PR TITLE
ENH: Add code to extract XML form data from Adobe ES4 PDF forms

### PR DESCRIPTION
**Pull request by georgealverson**

### CODE REVIEW

File: PyPDF2/pdf.py [hunk 1]
```diff
@@ -1272,11 +1272,12 @@ def getFields(self, tree = None, retval = None, fileobj = None):
...
         :rtype: dict, or ``None`` if form data could not be located.
         """
+        import zlib
         fieldAttributes = {"/FT" : "Field Type", "/Parent" : "Parent",
                        "/T" : "Field Name", "/TU" : "Alternate Field Name",
                        "/TM" : "Mapping Name", "/Ff" : "Field Flags",
-                       "/V" : "Value", "/DV" : "Default Value"}
-        if retval is None:
+                       "/V" : "Value", "/DV" : "Default Value",  "/XFA" : "XML form"}
+        if retval == None:
```
Comment (MartinThoma) [inline PyPDF2/pdf.py:1280]:
```suggestion
        if retval is None:
```

**Comment by MartinThoma** [comment]
@georgealverson Do you have an example PDF which we could use in the unit tests?

**Comment by georgealverson** [comment]
Dear Martin,

Attached is a PDF file using XFA. Below is a screenshot of the XFA content.

Best regards,
George

***@***.***

**Comment by MartinThoma** [comment]
Please either send it to info@martin-thoma.de or upload it directly on github. I cannot see an attachment here. I guess github filters it

**Comment by georgealverson** [comment]
[PyPDF_test_file.pdf](https://github.com/py-pdf/PyPDF2/files/8507185/PyPDF_test_file.pdf)
This is the file that was stripped

File: PyPDF2/pdf.py [hunk 2]
```diff
@@ -1297,12 +1298,22 @@ def getFields(self, tree = None, retval = None, fileobj = None):
                 self._buildField(tree, retval, fileobj, fieldAttributes)
                 break
 
-        if "/Fields" in tree:
-            fields = tree["/Fields"]
-            for f in fields:
-                field = f.getObject()
-                self._buildField(field, retval, fileobj, fieldAttributes)
-
+        if "/XFA" in tree:
```
Comment (MartinThoma) [inline PyPDF2/pdf.py:1301]: Why did you remove the logic around `"/Fields"`?

**Comment by georgealverson** [comment]
At this point, I no longer recall why that was done.

**Comment by MartinThoma** [comment]
Thank you for the swift response!

**Comment by MartinThoma** [comment]
I'm closing the PR here and continue the work in https://github.com/py-pdf/PyPDF2/pull/1026. I will list you as a co-author of that PR :-)

**Comment by georgealverson** [comment]
Dear Martin,

Sounds good. Thanks!

—George
